skrifa/src/outline/glyf/hint/definition.rs

```diff
@@ -0,0 +1,230 @@
+//! Management of function and instruction definitions.
+
+use core::ops::Range;
+
+use super::{error::HintErrorKind, program::Program};
+
+/// Code range and properties for a function or instruction definition.
+// Note: this type is designed to support allocation from user memory
+// so make sure the fields are all tightly packed and only use integral
+// types.
+// See <https://gitlab.freedesktop.org/freetype/freetype/-/blob/57617782464411201ce7bbc93b086c1b4d7d84a5/src/truetype/ttobjs.h#L158>
+#[derive(Copy, Clone, PartialEq, Eq, Default, Debug)]
+#[repr(C)]
+pub struct Definition {
+    start: u32,
+    end: u32,
+    // Either function number or opcode
```

Comment:
suggest comment to clarify that, iiuc, type is based on program

Reply:
The type is based on whether the definition is generated by FDEF or IDEF instructions. Both can occur in fpgm or prep. Will clarify in the comment.